engine/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -0,0 +1,65 @@
+package cromwell.services
+
+import akka.actor.Actor
+import cromwell.engine.workflow.WorkflowManagerActor.WorkflowMetadata
+import cromwell.engine.{ExecutionEventEntry, WorkflowId}
+import cromwell.engine.backend.JobKey
+import cromwell.services.MetadataServiceActor._
+import cromwell.webservice.WorkflowMetadataResponse
+import org.joda.time.DateTime
+
+object MetadataServiceActor {
+
+  sealed trait MetadataServiceActorMessage
+  case class SetStartTime(jobKey: JobKey, startTime: DateTime)
+  case class SetEndTime(jobKey: JobKey, time: DateTime)
+  case class AddExecutionEvent(jobKey: JobKey, executionEventEntry: ExecutionEventEntry)
+  /*
+  We'd have a "set" message for each of these:
```

Comment:
I think this is a different model - I'll see if I can mock it up in a separate PR so we can see them side by side